grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -662,12 +831,27 @@ def _GetPathArgsFromPath(path: str) -> List[str]:
 
 
 def _GetTypeName(
-    cls: Union[Descriptor, FieldDescriptor, EnumDescriptor, Type, int, str],
+    cls: Optional[
+      Union[Descriptor, FieldDescriptor, EnumDescriptor, Type, int, str]
```

Comment:
This type is used twice (here and in the `CreateSchema` method). Would it make sense to define and use an alias for it?

Reply:
It makes sense. Thank you for pointing this out! I introduced the alias named `TypeHinter` because the types in the union are either representing the type directly (such as `Descriptor`, `EnumDescriptor` and `Type)`, or identify a type through a label (such as an `int` `protobuf.TYPE_*` constant or a `str` represeting the name of the type for `BinaryStream`), or hold a type descriptor in their structure (`FieldDescriptor`) and "Hinter" is the "least generic" name I came up with. If you have a better suggestion for the alias name, please let me know.